app/views/building_map/index.html.erb

```diff
@@ -1,11 +1,141 @@
+
+<%= map(
     center: {
-        # HPI coordinates if no center is set
-        latlng: !@target.nil? ? @target.split(",") : [52.39339, 13.13208],
-        zoom: 18,
+        # HPI coordinates
+        latlng: RoutingHelper.center(@start),
+        zoom: 17,
     },
-    polygons: BuildingMapHelper.leaflet_polygons(),
+    polygons: BuildingMapHelper.leaflet_polygons,
     polylines: BuildingMapHelper.leaflet_polylines(@route),
     markers: BuildingMapHelper.leaflet_markers(@route, @target))
 %>
+
+<script>
+    let currentLocation;
+
+	function validate_start_input() {
+		const start_input = document.getElementById("start_input");
+		const startOptions = document.getElementById("startOptions").options;
+		let valid_start = false;
+		for (option of startOptions) {	
+			if (option.value === start_input.value) valid_start = true;
+		}
+		return valid_start;	
+	}
+
+	function validate_dest_input() {
+		const dest_input = document.getElementById("dest_input");
+		const destOptions = document.getElementById("destOptions").options;
+		let valid_dest = false;
+		for (option of destOptions) {	
+			if (option.value === dest_input.value) valid_dest = true;
+		}
+		return valid_dest;
+	}
+
+	const start_input_field = document.getElementById("start_input");
+	start_input_field.setCustomValidity("Please select a valid starting place.");
+	start_input_field.addEventListener("change", () => {
+		if (validate_start_input()) {
+			start_input_field.setCustomValidity("");
+		} else {
+			start_input_field.setCustomValidity("Please select a valid starting place.");
+		}
+	});
+
+	const dest_input_field = document.getElementById("dest_input");
+	dest_input_field.setCustomValidity("Please select a valid destination place.");
+	dest_input_field.addEventListener("change", () => {
+		if (validate_dest_input()) {
+			dest_input_field.setCustomValidity("");
+		} else {
+			dest_input_field.setCustomValidity("Please select a valid destination place.");
+		}
+	});
+
+	function request_location() {
+		if (start_input_field.value !== "Your location") return;
+		navigator.geolocation.getCurrentPosition(
+			(pos) => {
+				currentLocation = String(pos.coords.latitude) + "," + String(pos.coords.longitude);
+				start_input_field.setCustomValidity("");
+			},  
+			(error) => {
+				console.warn(`ERROR(${error.code}): ${error.message}`);
+				switch(error.code) {
+					case GeolocationPositionError.TIMEOUT:
+						start_input_field.setCustomValidity("Your browser could not determine your position. Please choose a different starting place.")
+						break;
+					case GeolocationPositionError.PERMISSION_DENIED:
+						start_input_field.setCustomValidity("You have to grant your browser the permission to access your location if you want to use this feature.")
+						break;
+					case GeolocationPositionError.POSITION_UNAVAILABLE:
+						start_input_field.setCustomValidity("Your browser could not determine your position. Please choose a different starting place.")
+						break;
```

Comment:
Since `TIMEOUT` and `POSITION_UNAVAILABLE` are the same, why not switch (or if-branch) over `case Geo...PERMISSION_DENIED` and `default`? Are there other errors that should be ignored?

Reply:
You're right, there are only [these](https://developer.mozilla.org/en-US/docs/Web/API/GeolocationPositionError) three possible error codes, so I changed this according to your suggestion.